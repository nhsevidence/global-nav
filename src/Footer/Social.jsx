import React, { Component } from "react";
import FacebookIcon from "@nice-digital/icons/lib/Facebook";
import TwitterIcon from "@nice-digital/icons/lib/Twitter";
import YouTubeIcon from "@nice-digital/icons/lib/YoutubePlay";
//import Instagram from "@nice-digital/icons/lib/Instagram";

import styles from "./Social.module.scss";

export default class Social extends Component {
	render() {
		return (
			<div className={styles.social}>
				<ul>
					<li>
						<a href="https://en-gb.facebook.com/NationalInstituteforHealthandCareExcellence/">
							{typeof SVGRect !== "undefined" && <FacebookIcon />}
							Facebook
						</a>
					</li>
					<li>
						<a href="https://twitter.com/NICEcomms">
							{typeof SVGRect !== "undefined" && <TwitterIcon />}
							Twitter
						</a>
					</li>
					<li>
						<a href="https://www.youtube.com/user/NICEmedia">
							{typeof SVGRect !== "undefined" && <YouTubeIcon />}
							YouTube
						</a>
					</li>
					<li>
						<a href="https://www.instagram.com/nicecomms/">
							{typeof SVGRect !== "undefined" && <YouTubeIcon />}
							Instagram
						</a>
					</li>
				</ul>
			</div>
		);
	}
}
