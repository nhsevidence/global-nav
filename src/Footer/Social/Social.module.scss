@import '~@nice-digital/nds-core/scss/core';

.wrapper {
  clear: left;
}

.list {
  list-style: none;
  margin: 0;
  padding: 0;

  li {
    margin: rem(0 0 $nds-spacing-x-small);
  }

  svg {
    margin: rem(-$nds-spacing-xx-small $nds-spacing-x-small 0 0);
    vertical-align: middle;
  }
}

@include mq($from: sm) {
  .list {
    li {
      float: left;
      margin: rem(0 $nds-spacing-medium 0 0);
    }
  }
}

@include mq($from: md) {
  .wrapper {
    clear: none;

    li {
      float: none;
      margin: rem(0 0 $nds-spacing-x-small);
    }
  }
}
