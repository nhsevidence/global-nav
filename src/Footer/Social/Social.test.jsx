import React from "react";

import Social from "./Social";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Social", () => {
	it("Renders without crashing", () => {
		const wrapper = shallow(<Social />);
		expect(wrapper).toHaveLength(1);
	});

	it("Matches snapshot with no SVG support", () => {
		const wrapper = shallow(<Social />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Matches snapshot with SVG support", () => {
		window.SVGRect = "anything";
		const wrapper = shallow(<Social />);
		expect(toJson(wrapper)).toMatchSnapshot();
		delete window.SVGRect;
	});
});
