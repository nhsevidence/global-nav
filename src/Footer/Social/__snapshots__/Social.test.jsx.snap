// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Matches snapshot with SVG support 1`] = `
<div
  className="wrapper"
>
  <ul
    aria-label="NICE on social media"
    className="list"
  >
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Facebook"
        href="https://en-gb.facebook.com/NationalInstituteforHealthandCareExcellence/"
      >
        <SvgFacebook />
        Facebook
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Twitter"
        href="https://twitter.com/NICEcomms"
      >
        <SvgTwitter />
        Twitter
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="YouTube"
        href="https://www.youtube.com/user/NICEmedia"
      >
        <SvgYoutubePlay />
        YouTube
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Instagram"
        href="https://www.instagram.com/nicecomms/"
      >
        <SvgInstagram />
        Instagram
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="LinkedIn"
        href="https://www.linkedin.com/company/national-institute-for-health-and-clinical-excellence"
      >
        <SvgLinkedin />
        LinkedIn
      </TrackedLink>
    </li>
  </ul>
</div>
`;

exports[`Social Matches snapshot with no SVG support 1`] = `
<div
  className="wrapper"
>
  <ul
    aria-label="NICE on social media"
    className="list"
  >
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Facebook"
        href="https://en-gb.facebook.com/NationalInstituteforHealthandCareExcellence/"
      >
        Facebook
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Twitter"
        href="https://twitter.com/NICEcomms"
      >
        Twitter
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="YouTube"
        href="https://www.youtube.com/user/NICEmedia"
      >
        YouTube
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="Instagram"
        href="https://www.instagram.com/nicecomms/"
      >
        Instagram
      </TrackedLink>
    </li>
    <li>
      <TrackedLink
        eventAction="Footer click"
        eventCategory="TopHat and footer"
        eventLabel="LinkedIn"
        href="https://www.linkedin.com/company/national-institute-for-health-and-clinical-excellence"
      >
        LinkedIn
      </TrackedLink>
    </li>
  </ul>
</div>
`;
